libyul/ObjectParser.cpp

```diff
@@ -104,9 +111,57 @@ shared_ptr<Block> ObjectParser::parseCode()
 	return parseBlock();
 }
 
+optional<ObjectParser::SourceNameMap> ObjectParser::tryParseSourceNameMapping() const
+{
+	// @use-src 0:"abc.sol", 1:"foo.sol", 2:"bar.sol"
+	//
+	// UseSrcList := UseSrc (',' UseSrc)*
+	// UseSrc     := [0-9]+ ':' FileName
+	// FileName   := "(([^\"]|\.)*)"
+
+	// Matches some "@use-src TEXT".
+	static std::regex const lineRE = std::regex(
+		"(^|\\s)@use-src(\\s+(.*))?$",
```

Comment:
We still have to test that `$` works correctly for e.g. `@use-src 0:"x.sol" @somethingelse bla`.

Maybe it is better to just construct the scanner for the full comment literal and than let it run until it finds an identifier `@use-src`.